src/Microsoft.ML.PipelineInference/Macros/PipelineSweeperMacro.cs

```diff
@@ -36,6 +36,30 @@ public sealed class Arguments
 
             [Argument(ArgumentType.AtMostOnce, HelpText = "Output datasets from previous iteration of sweep.", SortOrder = 7, Hide = true)]
             public IDataView[] CandidateOutputs;
+
+            [Argument(ArgumentType.MultipleUnique, HelpText = "Column(s) to use as Role 'Label'", SortOrder = 8)]
+            public string[] LabelColumns;
+
+            [Argument(ArgumentType.MultipleUnique, HelpText = "Column(s) to use as Role 'Group'", SortOrder = 9)]
+            public string[] GroupColumns;
+
+            [Argument(ArgumentType.MultipleUnique, HelpText = "Column(s) to use as Role 'Weight'", SortOrder = 10)]
+            public string[] WeightColumns;
+
+            [Argument(ArgumentType.MultipleUnique, HelpText = "Column(s) to use as Role 'Name'", SortOrder = 11)]
+            public string[] NameColumns;
+
+            [Argument(ArgumentType.MultipleUnique, HelpText = "Column(s) to use as Role 'NumericFeature'", SortOrder = 12)]
+            public string[] NumericFeatureColumns;
+
+            [Argument(ArgumentType.MultipleUnique, HelpText = "Column(s) to use as Role 'CategoricalFeature'", SortOrder = 13)]
+            public string[] CategoricalFeatureColumns;
+
+            [Argument(ArgumentType.MultipleUnique, HelpText = "Column(s) to use as Role 'TextFeature'", SortOrder = 14)]
+            public string[] TextFeatureColumns;
+
+            [Argument(ArgumentType.MultipleUnique, HelpText = "Column(s) to use as Role 'ImagePath'", SortOrder = 15)]
+            public string[] ImagePathColumns;
```

Comment:
Shall we also log a bug about finding a more scalable solution?

Reply:
Yes, there is a bug for that already

---
In reply to: [202428704](https://github.com/dotnet/machinelearning/pull/461#discussion_r202428704) [](ancestors = 202428704)